app/services/user_merger.rb

```diff
@@ -0,0 +1,339 @@
+class UserMerger
+  def initialize(source_user, target_user)
+    @source_user = source_user
+    @target_user = target_user
+  end
+
+  def merge!
+    move_posts
+    update_user_ids
+    merge_given_daily_likes
+    merge_post_timings
+    merge_user_visits
+    update_site_settings
+
+    DiscourseEvent.trigger(:merging_users, @source_user, @target_user)
+    update_user_stats
+
+    delete_source_user
+    delete_source_user_references
+  end
+
+  protected
+
+  def move_posts
+    posts = Post.with_deleted
+      .where(user_id: @source_user.id)
+      .order(:topic_id, :post_number)
+      .pluck(:topic_id, :id)
+
+    topic_id = nil
+    post_ids = []
+
+    posts.each do |post|
```

Comment:
Instead of using `post[0]` and `post[1]` you can do

```ruby
posts.each do |(topic_id, post_id)|
```

Reply:
I think 

```
posts.each do |topic_id, post_id|
```

works too.

Reply:
Yeah, that works too. Thanks for the tip! 🙂